lib/src/platform/platform.dart

```diff
@@ -0,0 +1,125 @@
+
+  static bool _enableSTDERR = false;
+
+  /// If set to [true], allows write of [Level.error] and
+  /// [Level.wtf] to `stderr` (if present in the runtime platform).
+  ///
+  /// Default to [false], to be backward compatible with previous versions.
+  /// of this package.
+  static bool get enableSTDERR => _enableSTDERR;
+
+  static set enableSTDERR(bool value) {
+    _enableSTDERR = value ?? false;
+  }
+
+  /// Returns a [ConsolePrinter] that is muted (won't print anything).
+  static ConsolePrinter get consoleMutedPrinter => (arg) {};
+
+  /// Returns a [ConsolePrinter] for log events.
+  static ConsolePrinter get consoleLogPrinter => getPlatformConsoleLogPrinter();
+
+  /// Returns a [ConsolePrinter] for info events.
+  static ConsolePrinter get consoleInfoPrinter =>
+      getPlatformConsoleInfoPrinter();
+
+  /// Returns a [ConsolePrinter] for warn events.
+  static ConsolePrinter get consoleWarnPrinter =>
+      getPlatformConsoleWarnPrinter();
+
+  /// Returns a [ConsolePrinter] for error events.
+  static ConsolePrinter get consoleErrorPrinter =>
+      getPlatformConsoleErrorPrinter();
+
+  /// Returns the [ConsolePrinter] for the [level].
+  static ConsolePrinter getConsolePrinter(Level level) {
+    switch (level) {
+      case Level.verbose:
+      case Level.debug:
+        return consoleLogPrinter;
+      case Level.info:
+        return consoleInfoPrinter;
+      case Level.warning:
+        return consoleWarnPrinter;
+      case Level.error:
+      case Level.wtf:
+        return consoleErrorPrinter;
+      case Level.nothing:
+        return consoleMutedPrinter;
+      default:
+        return consoleLogPrinter;
+    }
+  }
+
+  /// In some browsers, depending of the [ConsolePrinter] for some [level],
+  /// the 1st line printed to the console is not vertically
+  /// aligned (head indented) with the other lines.
+  static bool isConsolePrinterHeadIndented(Level level) =>
+      isPlatformConsolePrinterHeadIndented(level);
+
+  /// Join lines, to avoid multiple calls to printer.
+  ///
+  /// In Web platform (browser), each call to printer, in case of `warnings` or
+  /// `erros`, will associate a stack trace tree (visible and expandable
+  /// in Browser console).
+  ///
+  /// In VM platform, multiple calls to printer is less efficient, since will
+  /// generate multiple IO calls.
+  static String joinLines(List<String> lines, Level level) {
+    if (lines == null) return '';
+    var linesLength = lines.length;
+    if (linesLength == 0) return '';
+
+    // Tries to avoid new [String] allocations:
+    String allLines;
+    if (linesLength == 1) {
+      allLines = lines[0];
+    } else {
+      var newLine = '\n';
+
+      // Ensure that head line is vertical aligned with other lines
+      // in the current console platform (if not, add a blank head line):
+      var buffer = StringBuffer(
+          LogPlatform.isConsolePrinterHeadIndented(level) ? newLine : '');
+
+      for (var i = 0; i < lines.length; i++) {
```

Comment:
I think you are reinventing the wheel here. See the implementation of [`join`](https://api.flutter.dev/flutter/dart-core/Iterable/join.html).

Reply:
The `join` version will create 1 full string and later append the header:
```
(LogPlatform.isConsolePrinterHeadIndented(level) ? '\n' : '') + lines.join('\n') ;
```
The `StringBuffer` version creates only the resulting string (`join` internally uses 'StringBuffer' too).

I did that since I think that logs should use low CPU and low memory, avoiding any overhead.

Reply:
I don't feel that the additional assignment is an issue warranting the additional code.

Reply:
I have made a "clean code" in this function.